packages/extension/src/providers/ethereum/types/evm-network.ts

```diff
@@ -103,10 +107,75 @@ export class EvmNetwork extends BaseNetwork {
   }
 
   public async getAllTokenInfo(address: string): Promise<AssetsType[]> {
+    const api = await this.api();
+    const tokensState = new TokensState();
+
+    const customTokens = await tokensState
+      .getTokensByNetwork(this.name)
+      .then((tokens) => {
+        const erc20Tokens = tokens.filter(
+          (token) => token.type === TokenType.ERC20
+        ) as CustomErc20Token[];
+
+        return erc20Tokens.map(
+          ({ name, symbol, address, icon, decimals }) =>
+            new Erc20Token({ name, symbol, contract: address, icon, decimals })
+        );
+      });
+
+    const balancePromises = customTokens.map((token) =>
+      token.getLatestUserBalance(api as API, address)
+    );
+
+    await Promise.all(balancePromises);
+
```

Comment:
you should use 
```
async getMarketInfoByContracts(
    contracts: string[],
    platformId: string
  ): Promise<Record<string, CoinGeckoTokenMarket | null>>
```
to get the market data, while you are at it can you create a new enum type for coingecko platform ids. 
https://github.com/enkryptcom/enKrypt/blob/e3f183d869a02af6cc378dfcb525ef211c4c0471/packages/extension/src/libs/market-data/types.ts#L7

and use that for in the network object instead of string. that way we can use that type for `platformId: string`

Reply:
But this is getting the user's token balance, not getting the token's price